app/Http/Controllers/OrderInvoice/EditController.php

```diff
@@ -108,24 +106,105 @@ public function update(Request $request)
                     $pnss_updated_at = now();
                     $payment_notice_sending = isset($data['payment_notice_sending']) ? 1 : 0;
 
-                    echo '9';
-
-                    for ($i = 0; $i < count($data['item']); $i ++) {
-                        echo '10';
-                        DB::table('invoices_table')->where('id', $invoice_id)->update([
-                            'item' => $data['item'][$i]
-                        ]);
+                    // 追加箇所
+                    // insertで追加されず、上書きされる
+                    foreach( $data['item_id'] as $item_id){
+                        var_dump($item_id); //1,0
+                    }
+                    //既にDBに入っているものがupdateされたあとにinsertされる
+                        foreach ($data['item'] as $item) {
```

Comment:
```suggestion
　　　　　　$item_data = [];
                        foreach ($data['item'] as $item) {
                        　　foreach ($data['item_id'] as $item_id) {
                        　　    $item_data[$item_id] = $item;
                        　　  }
                        　}
```
こんな感じに配列整形してからやると良いかも。
1ループで済むので。